setup.py

```diff
@@ -20,33 +20,36 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 # =============================================================================
 #
-
+import os
+import sys
+import platform
 from setuptools import setup, Extension
 from setuptools.command.build_ext import build_ext
-import sysconfig
-import platform
+from distutils import log
+
 
 with open("README.md", "r") as file_long_description:
     long_description = file_long_description.read()
 
-BUILD_DIR = "build"
-machine = platform.machine().lower()  # TODO: this is unreliable, need something better to detect target architecture
-is_armv7 = machine.startswith("armv7")  # TODO: can armv8_ still be AArch32?
-is_arm = is_armv7 or machine.startswith("arm") or machine.startswith("aarch64")
-is_x86 = machine in ["i386", "i686", "x86", "x86_64", "x64", "amd64"]
-
 
 class SAByEncBuild(build_ext):
     def build_extension(self, ext):
-        # determine compiler flags
-        compiler = self.compiler
-        if compiler.compiler_type == "msvc":
-            # LTCG not enabled due to issues seen with code generation where different ISA extensions are selected for specific files
-            base_ldflags = ["/OPT:REF", "/OPT:ICF"]
-            cflags = ["/O2", "/GS-", "/Gy", "/sdl-", "/Oy", "/Oi"]
+        # Try to determine the architecture to build for
+        machine = platform.machine().lower()
+        IS_ARM = machine.startswith("arm") or machine.startswith("aarch64")
+        IS_ARM7 = machine.startswith("armv7")
+        IS_X86 = machine in ["i386", "i686", "x86", "x86_64", "x64", "amd64"]
+        log.info("Detected: ARM=%s, ARM7=%s, X86=%s", IS_ARM, IS_ARM7, IS_X86)
+
+        # Determine compiler flags
+        if self.compiler.compiler_type == "msvc":
+            # LTCG not enabled due to issues seen with code generation where
+            # different ISA extensions are selected for specific files
+            ldflags = ["/OPT:REF", "/OPT:ICF"]
+            cflags = ["/O2", "/GS-", "/GL-", "/Gy", "/sdl-", "/Oy", "/Oi"]
         else:
             # TODO: consider -flto - may require some extra testing
-            base_ldflags = sysconfig.get_config_var("LDFLAGS").split()
```

Comment:
@animetosho I removed this part, because this picks up the FLAGS that Python itself was compiled with.
On cross-compiled version, this will reference the locations on the Python-build-system, and not the current-system-locations.
Setuptools automatically already adds almost all the parameters below. You can see that for example here, where many parameters are listed twice:
https://github.com/sabnzbd/sabyenc/runs/5332039624?check_suite_focus=true#step:5:354
Just wondering why you added the `sysconfig.get_config_var("LDFLAGS")` part? Maybe I'm missing something :)

Reply:
Oh nice.

I never checked - [the example](https://pythonextensionpatterns.readthedocs.io/en/latest/compiler_flags.html#setting-flags-automatically-in-setup-py) provided used it, so I went along (since most compilers don't mind flags being duplicated).